test/unit/medium_providers/plugin_test.rb

```diff
@@ -0,0 +1,28 @@
+require 'test_helper'
+
+class PluginMediumProviderTest < ActiveSupport::TestCase
+  class PluginMediumProvider < MediumProviders::Provider
+    def medium_uri(path = "", &block)
+      '/medium'
+    end
+
+    def valid?
+      true
+    end
+  end
+
+  setup do
+    Foreman::Plugin.medium_providers.register(PluginMediumProvider)
+  end
+
+  teardown do
+    Foreman::Plugin.medium_providers.unregister(PluginMediumProvider)
+  end
```

Comment:
Does `teardown` get called even in case of exception? I hope so.

Reply:
It does